src/dmd/cparse.d

```diff
@@ -2402,6 +2437,26 @@ final class CParser(AST) : Parser!AST
         }
     }
 
+    /*************************
+     * Simple __asm__ parser
+     * https://gcc.gnu.org/onlinedocs/gcc/Asm-Labels.html
+     * simple-asm-expr:
+     *   asm ( asm-string-literal )
+     *
+     * asm-string-literal:
+     *   string-literal
+     */
+    private AST.Expression cparseAsmLabel()
+    {
+        nextToken();     // move past __asm__
+        check(TOK.leftParenthesis);
+        if (token.value != TOK.string_)
+            error("string literal expected");
```

Comment:
Shouldn't the error be checked for *after* cparsePrimaryExp() ? I'm also concerned about the conflation between the grammar, which says `string-literal`, and the implementation, which parses a `primary-expression`. Should be consistent.

Reply:
Copied from `cparseStaticAssert`, which also has `string-literal` in the grammar.